src/backend/marsha/core/utils/xmpp_utils.py

```diff
@@ -0,0 +1,177 @@
+"""Utils for XMPP server."""
+from datetime import timedelta
+
+from django.conf import settings
+from django.utils import timezone
+
+import jwt
+import xmpp
+
+
+"""
+The XEP used to manage a Multi User Chat is XEP-0045 aka MUC.
+Spec are available at https://xmpp.org/extensions/xep-0045.html
+"""
+
+
+def _connect():
+    """Connect to a XMPP server and return the connection.
+
+    Returns
+    -------
+    xmpp.Client
+        A xmpp client authenticated to a XMPP server.
+    """
+    jid = xmpp.protocol.JID(settings.XMPP_PRIVATE_SERVER_JID)
+
+    client = xmpp.Client(server=jid.getDomain(), port=settings.XMPP_PRIVATE_SERVER_PORT)
+    client.connect()
+    client.auth(
+        user=jid.getNode(),
+        password=settings.XMPP_PRIVATE_SERVER_PASSWORD,
+        resource=jid.getResource(),
+    )
+
+    return client
+
+
+def create_room(room_name):
+    """Create and configure a room.
+
+    Documentation to create and configure a room:
+    https://xmpp.org/extensions/xep-0045.html#createroom
+
+    Parameters
+    ----------
+    room_name: string
+        The name of the room you want to create.
+    """
+    client = _connect()
+
+    client.send(
+        xmpp.Presence(
+            to=f"{room_name}@{settings.XMPP_CONFERENCE_URL}/admin",
+            payload=[xmpp.Node(tag="x", attrs={"xmlns": xmpp.NS_MUC})],
+        )
+    )
+
+    client.send(
```

Comment:
Just a question here.
As far as I understand, you are creating a "reserved room" with custom settings, right ?

The XEP-0045 [specifies](https://xmpp.org/extensions/xep-0045.html#createroom-reserved) that in this case : 
> the room owner **MUST** request an initial configuration form by sending an IQ get to the <room@service> itself containing an empty <query/> element qualified by the 'http://jabber.org/protocol/muc#owner' namespace:

This step seems to be skipped here, isn't it mandatory?

Reply:
I tried with and without this step and I see no difference. Maybe it's specific to prosody server and another XMPP server will follow strictly the spec.

Reply:
Ok thanks.
Since this PR is strongly linked to prosody because of the required modules, it's ok for me!